quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HttpPropertyMappers.java

```diff
@@ -127,6 +130,52 @@ public static PropertyMapper<?>[] getHttpPropertyMappers() {
 
     public static void validateConfig() {
         boolean enabled = isHttpEnabled(Configuration.getOptionalKcValue(HttpOptions.HTTP_ENABLED.getKey()));
+        boolean trustStoreFile = Configuration.getOptionalKcValue(HttpOptions.HTTPS_TRUST_STORE_FILE.getKey()).isPresent();
+        boolean keyStoreFile = Configuration.getOptionalKcValue(HttpOptions.HTTPS_KEY_STORE_FILE.getKey()).isPresent();
+
+        if (trustStoreFile) {
+            CertificateConfig config = new CertificateConfig();
+
+            config.trustStoreFile = Configuration.getOptionalKcValue(HttpOptions.HTTPS_TRUST_STORE_FILE.getKey()).map(Paths::get);
+            config.trustStorePassword = Configuration.getOptionalKcValue(HttpOptions.HTTPS_TRUST_STORE_PASSWORD.getKey());
+            config.trustStoreFileType = Configuration.getOptionalKcValue(HttpOptions.HTTPS_TRUST_STORE_TYPE.getKey());
+            config.trustStoreProvider = Configuration.getOptionalValue("quarkus.http.ssl.certificate.trust-store-provider");
+            config.trustStoreCertAlias = Configuration.getOptionalValue("quarkus.http.ssl.certificate.trust-store-cert-alias");
+            config.trustStoreFiles = Optional.empty();
+
+            try {
+                TlsUtils.computeTrustOptions(config, config.trustStorePassword);
+            } catch (IOException e) {
+                throw new PropertyException(Messages.httpsConfigurationNotSet());
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Unable to determine 'https-trust-store-type' automatically. " +
```

Comment:
I'm not sure about this exception message as it might not be only relevant to the file extension, right? There might be different causes of the failure, and the user should be notified about it as only this message might be confusing. See https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/TlsUtils.java#L34

@vmuzikar Or did I miss something?

Reply:
@mabartos Thank you for the review! Just for the context, these changes are cherry picked from @Pepo48's https://github.com/keycloak/keycloak/commit/9a4bcf8d7d2be995916eb96d9e4fd9f3faabf3e7.

I assume `TlsUtils` are called at multiple places, thus the user should see the proper exception as well. But I didn't double check.

Possibly we could rephrase a bit to something like:
_Unable to determine 'https-trust-store-type' automatically. Possible reason is missing file extension or the type property._
WDYT?